js/app.js

```diff
@@ -144,12 +158,20 @@ function randomizeMap() {
 
 function randomizePlayerSetup() {
   const players = Array.from(State.playerList);
-  const factions = randomizeFactions();
+  const useBot = document.getElementById("use-bot").checked;
```

Comment:
For consistency, I think you should set up a listener on the checkbox and update the State when it changes, saving it locally. That way when users load the page they get saved player lists and also bot choices, instead of only saving part of their inputs. It would also clean up this function so that it doesn't access the DOM and instead focuses purely on logic, while the listener handles the view-state interface.

Better yet, make this an integer, "number of bots", then randomly pick up to that number!

Reply:
Clicking `randomize` should not change state on whether something is selected or not. Randomization should be 'functional' based on provided state, and allowed to fail if the input state is invalid.  We fail if we provide too many players (which is reasonable), and it seems clearer to fail if we try to generate for one player (which is current behavior). 

I think there's a clearer way to represent this entire idea: have an "Add Bot" button at the same level as "Add Player" which could add a bot into the player list. Easy for user to add the desired number of players (0, 1, 2, 3...bots) and remove them. Bot players can be differentiated with color and a [BOT] tag. This means we can leverage the existing visual structure to communicate selected player state.

Faction selection for the bots would happen randomly from the currently available bot factions. This allows us to not have to modify this state as weirdly - randomizeFactions now just takes `two arguments` and doesn't need to change weird UI state.

In my opinion, this will be the cleanest way to allow users to add bots as it will also allow users to customize bots counts when playing with other players too.

Reply:
I'm fine with reworking the box into "add bot player" button in a separate PR that leverages what this uses, but I would love to hear your opinions. @tylerhuntley @haswellr 

I do feel strongly about not changing state here though